syntax = "proto2";

package msgTest;

message UserInfo {
    message Position {  
        optional float x = 1 [default = 0];  
        optional float y = 2 [default = 0];  
        optional float z = 3 [default = 0];  
    }

    message Velocity {  
        optional float x = 1 [default = 0];  
        optional float y = 2 [default = 0];  
        optional float z = 3 [default = 0]; 
    }

    required int32 id = 1;
    required Position position = 2; 
    required Velocity velocity = 3; 
}

message CS_Send_User_Info{
	required UserInfo userInfo = 1;
}

message CS_Request_User_Info{

}

message CS_Request_Other_User_Info{

}

message SC_Response_User_Info{
	required UserInfo userInfo = 1;
}

message SC_Response_Other_User_Info{
    repeated UserInfo usersInfo = 1;
}

message SC_Broadcast_User_Info{
	repeated UserInfo usersInfo = 1;
}

message SC_Add_User{
	required UserInfo userInfo = 1;
}